notebooks/Time_Series_Forecasting.ipynb

```diff
@@ -0,0 +1,660 @@
+   "metadata": {},
+   "source": [
+    "<i>Store id doesn't help itself with the sales predictions, so it is necessary to replace the store id with the corresponding information about it</i>"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "train_data = train_data.merge(stores_data, on=\"store_nbr\", how=\"left\")\n",
+    "train_data = train_data.drop('store_nbr', axis=1)\n",
+    "train_data.head()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "print('Number of \\'city\\', \\'state\\', \\'type\\' unique values: \\n')\n",
+    "for feature in ['city', 'state', 'type']:\n",
+    "    print(feature, len(train_data[feature].unique()))"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "plt.figure(figsize=(25,9))\n",
+    "sns.countplot(x=train_data['city'], alpha=0.7, data=train_data)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "plt.figure(figsize=(25,9))\n",
+    "sns.countplot(x=train_data['state'], alpha=0.7, data=train_data)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "<i>It seems that the presence of both features 'state' and 'city' isn't necessary, that is why we should delete one of them.<br>I'll choose 'state' feature, because 'city' feature gives us more information (there can be few cities in the state).<br>'city' feature has only 7 more values than 'state' feature, that is why speaking about the model complexity, there should not be much difference.</i>"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "train_data = train_data.drop('state', axis=1)\n",
+    "train_data.head()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "train_data_copy = train_data.copy()\n",
+    "train_data_copy.head()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "plt.figure(figsize=(20, 100))\n",
+    "for i, family_value in zip(range(len(family_values)), family_values):\n",
+    "  plt.subplot(17, 2, i+1)\n",
+    "  for city in train_data_copy['city'].unique():\n",
+    "      average_sales = train_data_copy[(train_data_copy['family'] == family_value) &\n",
+    "                                 (train_data_copy['city'] == city)].groupby('date').mean()['sales']\n",
+    "      trend = average_sales.rolling(\n",
+    "          window=365,\n",
+    "          center=True,\n",
+    "          min_periods=183,\n",
+    "      ).mean()\n",
+    "      ax = average_sales.plot(alpha=0)\n",
+    "      ax = trend.plot(ax=ax, linewidth=3)\n",
```

Comment:
I think some legend is required here to understand which line corresponds to which city.
```suggestion
    "      trend.plot(ax=ax, linewidth=3, label=city)\n",
    "      handles, labels = ax.get_legend_handles_labels()\n",
```
And at the end of your cell you can add
```
fig.legend(handles, labels, loc='lower center')
```
`fig` is the value returned by `plt.figure`. To be honest, I did not try this code, just read the [stackoverflow post](https://stackoverflow.com/questions/9834452/how-do-i-make-a-single-legend-for-many-subplots-with-matplotlib) (but hope it works)